### PR TITLE
[Web][SIMS-#920] Add button for Apply for PD

### PR DESCRIPTION
**Pull request by henryoforeh-aot**
I created a modal popup that displays when a user applies for PD Status. 
I removed the banner that used to display PD status .
I created a read-only field in the Student Information page to display Permanent Disability Status.

studentpdstatusmodal

studentpdstatus

### CODE REVIEW

File: sources/packages/web/src/components/students/modals/PDStatusApplicationModal.vue [hunk 1]
```diff
@@ -0,0 +1,60 @@
+<template>
+  <Dialog
```
Comment (ann-aot) [inline sources/packages/web/src/components/students/modals/PDStatusApplicationModal.vue:2]: mm, henry, as I shared we do have a generic component for modal called `ModalDialogBase`. you can refer the same example i shared 
image

Comment (ann-aot) [reply sources/packages/web/src/components/students/modals/PDStatusApplicationModal.vue:2]: this is primevue, we are using vuetify modal

**Comment by andrewsignori-aot** [comment]
Is there any specific reason why the button on the UI could not be more like what we have on Figma?
image

File: sources/packages/web/src/components/students/modals/PDStatusApplicationModal.vue [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  components: {
+    ModalDialogBase,
+  },
+  props: {
+    showModal: {
+      type: Boolean,
+      required: true,
+    },
+  },
+  setup(props: any, context: any) {
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/components/students/modals/PDStatusApplicationModal.vue:37]: The code to control the dialog can be simplified here and on the consumer. We can talk about it.
Please see below how the code would looks like in the modal.
image

And what the code will look like in the consumer.
```
 <PDStatusApplicationModal ref="pdStatusApplicationModal" />
```
image

File: sources/packages/web/src/components/students/modals/PDStatusApplicationModal.vue [hunk 3]
```diff
@@ -0,0 +1,44 @@
+<template>
+  <ModalDialogBase
+    title="Apply for a permanent disability status"
+    :showDialog="showDialog"
+  >
+    <template v-slot:content>
+      <v-container>
```
Comment (ann-aot) [inline sources/packages/web/src/components/students/modals/PDStatusApplicationModal.vue:7]: do we need `v-container` here

File: sources/packages/web/src/components/students/modals/PDStatusApplicationModal.vue [hunk 4]
```diff
@@ -0,0 +1,44 @@
+<template>
+  <ModalDialogBase
+    title="Apply for a permanent disability status"
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/components/students/modals/PDStatusApplicationModal.vue:3]: Please added the dialogType="question" also.

File: sources/packages/web/src/components/students/modals/PDStatusApplicationModal.vue [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+import ModalDialogBase from "@/components/generic/ModalDialogBase.vue";
+import { useModalDialog } from "@/composables";
+
+export default {
+  components: {
+    ModalDialogBase,
+  },
+  setup() {
+    const { showDialog, showModal, resolvePromise } = useModalDialog<boolean>();
+    const requestPD = async (payload: boolean) => {
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/components/students/modals/PDStatusApplicationModal.vue:32]: I believe that the async can be removed here.

File: sources/packages/web/src/views/student/StudentProfile.vue [hunk 6]
```diff
@@ -126,31 +98,55 @@ export default {
...
-        showApplyPDButton.value = true;
+    const getStudentDetails = async () => {
+      const studentRestriction = await StudentService.shared.getStudentRestriction();
+      hasRestriction.value = studentRestriction.hasRestriction;
+      restrictionMessage.value = studentRestriction.restrictionMessage;
+      if (props.editMode) {
+        await getStudentInfo();
+        const data: StudentFormData = {
+          ...studentAllInfo.value,
+          ...studentAllInfo.value.contact,
```
Comment (ann-aot) [inline sources/packages/web/src/views/student/StudentProfile.vue:109]: not part of this PR, with lines 108 and 109, data will be something like 
image
is this the expected result? @andrewsignori-aot  @guru-aot @dheepak-aot

**Review by ann-aot — COMMENTED** [review]
LGTM, just some minor comments

**Review by ann-aot — APPROVED** [review]
Thanks for doing the change 👍 Good Work @henryoforeh-aot 😄

File: sources/packages/web/src/views/student/StudentProfile.vue (hunk 6 above)
Comment (andrewsignori-aot) [reply sources/packages/web/src/views/student/StudentProfile.vue:109]: As we talked about in the call, this is copying the contact data to the object root because the current form.io definition is expected this strcutuire.
In the future, we can avoid this using the container component from form.io to create a child object.
image

**Review by andrewsignori-aot — APPROVED** [review]
Thanks for doing the changes, nice work, looks good 👍

**Review by guru-aot — APPROVED** [review]
LGTM, thanks @henryoforeh-aot for incorporating the changes 👍

**Review by dheepak-aot — APPROVED** [review]
Good work @henryoforeh-aot . 👍
